src/db.rs

```diff
@@ -96,4 +109,28 @@ impl Database {
     pub fn insert_witnesses(&self) -> Result<()> {
         unimplemented!();
     }
+
+    pub fn create_database_if_necessary(&self) -> Result<()> {
```

Comment:
this should not be a member function, but a global public function.
at the beginning we don't even have a database connection.